src/main/java/hudson/plugins/spotinst/cloud/monitor/AwsSpotinstCloudInstanceTypeMonitor.java

```diff
@@ -0,0 +1,81 @@
+package hudson.plugins.spotinst.cloud.monitor;
+
+import hudson.Extension;
+import hudson.model.AdministrativeMonitor;
+import hudson.plugins.spotinst.cloud.AwsSpotinstCloud;
+import hudson.slaves.Cloud;
+import jenkins.model.Jenkins;
+import org.apache.commons.collections.CollectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Extension
+public class AwsSpotinstCloudInstanceTypeMonitor extends AdministrativeMonitor {
+    //region members
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsSpotinstCloud.class);
+    Map<String, List<String>> invalidInstances;
+    //endregion
+
+    //region Overrides
+    @Override
+    public boolean isActivated() {
+        boolean retVal;
+        invalidInstances = new HashMap<>();
```

Comment:
extract to method : initinvalidInstances

Reply:
Done